packages/connect/src/connector/smoldot-light.ts

```diff
@@ -22,26 +22,68 @@ const getStart = () => {
   return startPromise
 }
 
-let clientNumReferences = 0
+const clientReferences: Config[] = [] // Note that this can't be a set, as the same config is added/removed multiple times
 let clientPromise: Promise<Client> | null = null
-const getClientAndIncRef = (): Promise<Client> => {
+let clientReferencesMaxLogLevel = 3
+const getClientAndIncRef = (config: Config): Promise<Client> => {
+  if (config.maxLogLevel && config.maxLogLevel > clientReferencesMaxLogLevel)
```

Comment:
I think this works same:
```suggestion
  if (config?.maxLogLevel > clientReferencesMaxLogLevel)
```

Reply:
I don't really like this change. You're comparing null with a number, which doesn't make sense and should be forbidden by TypeScript if it was me.

Reply:
That is not a comparison of null with a number. The `if (config?.maxLogLevel > clientReferencesMaxLogLevel)` actually translates to `if ((config === null || config === void 0 ? void 0 : config.maxLogLevel) > clientReferencesMaxLogLevel)` so its actual a comparison between numbers, but its just a detail and doesn't really matter which one we use. Result is same